Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/SharedOpaqueDirectoryTests.cs

```diff
@@ -976,23 +1041,24 @@ public void CacheConvergenceCanTriggerAdditionalOutputContentAwareViolations()
         }
 
         private void FirstDoubleWriteWinsMakesPipUnCacheableRunner(
-            IEnumerable<Operation> writeOperation, 
+            IEnumerable<Operation> writeOperations, 
             AbsolutePath sharedOpaqueDirPath, 
             bool originalProducerPipCacheHitExpected,
             bool doubleWriteProducerPipCacheHitExpected)
         {
             // originalProducerPip writes an artifact in a shared opaque directory
-            ProcessBuilder originalProducerPipBuilder = CreatePipBuilder(writeOperation);
+            ProcessBuilder originalProducerPipBuilder = CreatePipBuilder(
+                writeOperations.Append(Operation.WriteFile(FileArtifact.CreateOutputFile(ObjectRootPath.Combine(Context.PathTable, "dep")))));
             originalProducerPipBuilder.AddOutputDirectory(sharedOpaqueDirPath, SealDirectoryKind.SharedOpaque);
             originalProducerPipBuilder.DoubleWritePolicy |= DoubleWritePolicy.UnsafeFirstDoubleWriteWins;
             ProcessWithOutputs originalProducerPipResult = SchedulePipBuilder(originalProducerPipBuilder);
 
             // doubleWriteProducerPip writes the same artifact in a shared opaque directory
-            ProcessBuilder doubleWriteProducerPipBuilder = CreatePipBuilder(writeOperation);
+            ProcessBuilder doubleWriteProducerPipBuilder = CreatePipBuilder(writeOperations);
             doubleWriteProducerPipBuilder.AddOutputDirectory(sharedOpaqueDirPath, SealDirectoryKind.SharedOpaque);
 
-            // Let's make doubleWriteProducerPip depend on originalProducerPip so we avoid potential file locks on the double write
-            doubleWriteProducerPipBuilder.AddInputDirectory(originalProducerPipResult.Process.DirectoryOutputs.Single());
+            // Let's make doubleWriteProducerPip depends on originalProducerPip so we avoid potential file locks on the double write

```

Comment:
depend

Reply:
The problem of reading only a fraction of the sentence :)

---
In reply to: [284502181](https://github.com/microsoft/BuildXL/pull/316#discussion_r284502181) [](ancestors = 284502181)